### PR TITLE
Add docs for maintainers and enforce hugo version

### PR DESCRIPTION
**Pull request by melissawm**
- Make sure the site builds with the scientific python hugo theme
- Draft documentation for maintainers
- Also creates .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.hugo_build.lock
+public/
+resources/_gen/
```
Comment (steppi) [inline .gitignore]:
```suggestion
resources/_gen/

```

**Comment by melissawm** [comment]
Expand on the fact that we have an extra layer of indirection throught the repos at scientific-python-translations org.

**Comment by melissawm** [comment]
Also, the Create translations PR targets the scientific-python-translations repos, NOT the final website repos (this is only true for NumPy.org)

File: content/docs.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Ideally you will **not** have to run this script directly, but can include it as
+part of your website deployment process (see, for example,
+[numpy/numpy.org#772](https://github.com/numpy/numpy.org/pull/772))
+
+{{< admonition important >}}
+To prevent future conflicts with the GitHub/Crowdin integration, it is important
+that you set your translations to duplicate strings to be an exact match:
+
+screenshot
```
Comment (melissawm) [inline content/docs.md]: I can't find the correct setting in Crowdin, @steppi would you help me complete this?

File: content/docs.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+<center><img alt="Screenshot of the Run workflow dialog from GitHub, with an input field labeled Crowding language code for the language of interest" src="../images/run_workflow.png" width=800/></center>
+
+After these steps, a PR will be created to your website repo with the
+translations for the language you selected (see
+[numpy/numpy.org#774](https://github.com/numpy/numpy.org/pull/774) for an example.) This PR should be
+merged when you are ready to publish the translations.
+
+After merging the translations PR, you should **delete the corresponding branch** and
+re-create it from scratch. This will prevent conflicts with future translations.
```
Comment (melissawm) [inline content/docs.md]: @steppi please confirm

File: content/docs.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+## Setting up a language switcher
+
+This work is in progress - follow (issue #) for details.
```
Comment (melissawm) [inline content/docs.md:118]: We should create an issue to update folks on how this is going.

File: content/docs.md (hunk 3 above)
Comment (steppi) [inline content/docs.md]:
```suggestion
After merging the translations PR, the Crowdin service branch (by default, named `l10n_main`) will have merge conflicts with `main`. To fix this, delete the Crowdin service branch. Crowdin will automatically recreate the service branch with merge conflicts resolved. This same process can also be used to resolve merge conflicts if translations are updated outside of Crowdin. (Setting the duplicate strings setting to *Hide (regular detection) - all duplicates will share the same translation* makes Crowdin better able to resolve merge conflicts without un-translating strings).
```

File: content/docs.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+   synced. By creating a new branch with only the commits we are interested in,
+   we can bypass this limitation.
+
+Ideally you will **not** have to run this script directly, but can include it as
+part of your website deployment process (see, for example,
+[numpy/numpy.org#772](https://github.com/numpy/numpy.org/pull/772))
+
+{{< admonition important >}}
+To prevent future conflicts with the GitHub/Crowdin integration, it is important
+that you set your translations to duplicate strings to be an exact match:
```
Comment (steppi) [inline content/docs.md]:
```suggestion
To prevent future conflicts with the GitHub/Crowdin integration, it is important
that you the duplicate strings setting under *Settings*, *import*, *Duplicates* to
*Hide (regular detection) - all duplicates will share the same translation*.
```

Also. Does it make sense to include a screenshot?

image

**Review by steppi — COMMENTED** [review]
Thanks @melissawm. I made some suggestions for the part about hiding duplicates, and the description for how to resolve merge conflicts.

File: content/docs.md (hunk 3 above)
Comment (melissawm) [reply content/docs.md]: Thank you!

File: content/docs.md (hunk 5 above)
Comment (melissawm) [reply content/docs.md]: Yes - more importantly can you share how you got to this menu option? I tried going through the settings but could not find it

Comment (melissawm) [reply content/docs.md]: Nevermind I found it!

**Comment by melissawm** [comment]
I think this is good to merge.

**Comment by melissawm** [comment]
Hi @goanpeca - would you take a look if you have some time? I think you would be the ideal reviewer as you are now onboarding into this process. Thank you!

File: content/docs.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+---
+title: "Documentation for maintainers"
```
Comment (trallard) [inline content/docs.md]: Nit, but I generally prefer persona-first like Maintainers' docs

File: content/docs.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+for project websites. As each project has their own set up, the patterns
+described here may not fit exactly, but they will help you understand the
+basic workflows.
+
+At least some of the tasks described here rely on scripts from
+https://github.com/Scientific-Python-Translations/automations/.
+
+## Integration with Crowdin
+
+Crowdin provides GitHub integration tools to sync all translated content from
```
Comment (trallard) [inline content/docs.md]: Link to Crowdin on first mention

File: content/docs.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+the Crowdin web interface into project websites and vice-versa. However, a few
+adaptations are needed.
+
+Given a GitHub repository which has been synced to Crowdin, the
```
Comment (trallard) [inline content/docs.md]:
```suggestion
Once a GitHub repository has been synced to Crowdin, the
```

Make it easier to follow/read

File: content/docs.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+[`create_branch_for_language.sh` script](https://github.com/Scientific-Python-Translations/automations/blob/main/scripts/create_branch_for_language.sh)
+is used to create a new branch including all (and only) commits from Crowdin's
+branch for a particular language of interest. This script is useful because:
+
+1. Crowdin will keep commits for all languages under translation in the same
+   branch. However, we prefer to be able to create, review, and merge pull
+   requests for one language at a time.
```
Comment (trallard) [inline content/docs.md]:
```suggestion
1. Crowdin will keep commits for all the available languages under translation in the same
   branch. However, we prefer to be able to create pull
   requests for one language at a time to keep Pull Request reviews smaller and easier to manage.
```

File: content/docs.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+2. There are times when translations must be edited manually (outside of
+   Crowdin) due to incorrect segmentation of the content into strings for
+   translation. Crowdin's branch can only be edited through the Crowdin UI, and
+   commits pushed to it outside of Crowdin will be lost when the branch is
+   synced. By creating a new branch with only the commits we are interested in,
+   we can bypass this limitation.
+
+Ideally you will **not** have to run this script directly, but can include it as
```
Comment (trallard) [inline content/docs.md]: Which script? the create branch one?

File: content/docs.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+This work is in progress - follow [pydata/pydata-sphinx-theme#507](https://github.com/pydata/pydata-sphinx-theme/issues/507) for details.
+
+## Known limitations
+
+### Missing translations
+
+For items such as news items and release announcements, translations may not
+always be up to date. In this case, your project can decide what to do with
+these items (for example, keep them in english or hide them from the deployed
+site.)
```
Comment (trallard) [inline content/docs.md]:
```suggestion
Translations may not always be up to date for items such as news items and release announcements. In this case, your project can decide what to do with
these items (for example, keep them in English or hide them from the deployed
site.)
```

**Review by trallard — COMMENTED** [review]
I am not sure what it is, but I found this a bit hard to follow, it almost feels like set actual steps needed by maintainers are intermingled with a lot of implementation/caveat details about Crowdin and how the integration is managed. 
Though, I do not have a concrete proposal as to how to improve the docs right now, perhaps keep a TLD;R or summary of what maintainers _actually_ need to do and then refer to the rest of the content as an extended guide?

File: content/docs.md (hunk 10 above)
Comment (melissawm) [reply content/docs.md]: Yes - added a note

**Comment by melissawm** [comment]
Ok - I've tried reorganizing a bit to make the steps clearer and the information about crowdin internals less intrusive.

File: content/docs.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+
+Ideally you will **not** have to run the `create_branch_for_language.sh` script
+directly, but can include it as part of your website deployment process (see,
+for example, [numpy/numpy.org#772](https://github.com/numpy/numpy.org/pull/772))
+
+{{< admonition important >}}
+To prevent future conflicts with the GitHub/Crowdin integration, it is important
+that you set your translations to duplicate strings to be an exact match. To do
+this, navigate to your project's Settings in Crowdin, select Import and under
+"Source strings" -> "Duplicates" choose "Hide (regular detection)".
```
Comment (steppi) [inline content/docs.md]:
```suggestion
To prevent future conflicts with the GitHub/Crowdin integration, it is important
that you configure Crowdin to have duplicate strings share the same translation.
To do this, navigate to your project's Settings in Crowdin, select Import and under
"Source strings" -> "Duplicates" choose "Hide (regular detection)".
```

Sorry, I think the language about an exact match is probably something I'd told you, but it turns out that what Crowdin means by an "exact match" is that the source strings match and the string IDs in the segmentation match. My understanding now is that "exact match"  is useful when you want translations to be shared for the same string across branches, but not the same string in different places within the same branch.

The suggestion here mirrors the language Crowdin uses to describe the feature.

**Review by steppi — COMMENTED** [review]
One minor suggestion. After that I think this is good to ship.

**Comment by goanpeca** [comment]
Hi @melissawm looks good to me!

After the other automations are in place I can update the docs to match :) 

---

Should we go ahead and merge?

**Review by goanpeca — APPROVED** [review]

**Comment by melissawm** [comment]
I think we can merge and follow up with any improvements. I'll work on the switcher docs next, and we can use these docs to follow the scipy.org deployment process.

Thank you!
